chef/cookbooks/crowbar/recipes/prepare-upgrade-scripts.rb

```diff
@@ -88,3 +88,25 @@
     use_ha: use_ha
   )
 end
+
+# Following script executes all actions that are needed directly on the node
+# directly before the OS upgrade is initiated.
+
+# Find out v2 auth URL that is needed for neutron-ha-tool
+keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
```

Comment:
As I'm using this only to get auth url for neutron-ha-tool, I assume I can limit the KeystoneHelper call to the nodes with neutron role(s) deployed. 
That way I'd prevent those errors I hit yesterday when retrieving keystone_settings failed. (That said, it should not fail anyway and the library should be fixed). @vuntz what do you think?